src/streaming/Stream.js

```diff
@@ -644,6 +702,80 @@ function Stream(config) {
         checkIfInitializationCompleted();
     }
 
+    function isCompatibleWithStream(stream) {
+        return compareCodecs(stream, Constants.VIDEO) && compareCodecs(stream, Constants.AUDIO);
```

Comment:
This will return false for audio-only (and video-only streams), which is incorrect.

Reply:
This requires a few changes like checking if the previous / next period has audio and video to avoid saving the non-needed buffer or recreating it, will add it last.

Reply:
I added commit d7f4b8f to allow preload of audio-only or video-only periods, but won't work when switching from a video/audio period to a video-only one

Reply:
Following that commit, this works great for multiperiod audio-only services :champagne: